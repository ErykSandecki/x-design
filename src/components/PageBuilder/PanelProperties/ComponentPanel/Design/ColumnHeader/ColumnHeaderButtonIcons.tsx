import { ReactNode } from 'react';

// components
import { UITools } from 'shared';

// others
import { TOOLTIP_TRANSLATION_KEY } from 'constant/constants';

const ColumnHeaderButtonIcons = (t: TT): Array<ReactNode> => [
  <UITools.ButtonIcon
    key={0}
    name="HtmlTag"
    tooltip={{ autoPositioning: true, content: t(`${TOOLTIP_TRANSLATION_KEY}.readyForDevStatus`) }}
  />,
  <UITools.ButtonIcon
    key={1}
    name="Component"
    tooltip={{ autoPositioning: true, content: t(`${TOOLTIP_TRANSLATION_KEY}.createComponent`) }}
  />,
];

export default ColumnHeaderButtonIcons;
