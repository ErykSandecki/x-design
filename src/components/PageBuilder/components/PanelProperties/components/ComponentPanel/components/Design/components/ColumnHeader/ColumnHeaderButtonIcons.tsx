import { ReactNode } from 'react';

// components
import { UITools } from 'shared';

const ColumnHeaderButtonIcons = (): Array<ReactNode> => [
  <UITools.ButtonIcon key={0} name="HtmlTag" />,
  <UITools.ButtonIcon key={1} name="Component" />,
];

export default ColumnHeaderButtonIcons;
