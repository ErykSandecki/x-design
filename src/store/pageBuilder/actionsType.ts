export const REDUCER_KEY = 'pageBuilder';

export const ADD_ELEMENT = `${REDUCER_KEY}/ADD_ELEMENT`;
export const APPLY_ELEMENTS_SIZE_TYPE = `${REDUCER_KEY}/APPLY_ELEMENTS_SIZE_TYPE`;
export const CHANGE_ALIGNMENT = `${REDUCER_KEY}/CHANGE_ALIGNMENT`;
export const CHANGE_BACKGROUND = `${REDUCER_KEY}/CHANGE_BACKGROUND`;
export const CHANGE_LAYOUT = `${REDUCER_KEY}/CHANGE_LAYOUT`;
export const CHANGE_PARENT = `${REDUCER_KEY}/CHANGE_PARENT`;
export const CHANGE_POSITION = `${REDUCER_KEY}/CHANGE_POSITION`;
export const CLEAR_PREV_STATE = `${REDUCER_KEY}/CLEAR_PREV_STATE`;
export const FIT_LAYOUT = `${REDUCER_KEY}/FIT_LAYOUT`;
export const FLIP_ELEMENTS = `${REDUCER_KEY}/FLIP_ELEMENTS`;
export const REDUCER_HISTORY_REDO = `${REDUCER_KEY}/REDUCER_HISTORY_REDO`;
export const REDUCER_HISTORY_SAVE = `${REDUCER_KEY}/REDUCER_HISTORY_SAVE`;
export const REDUCER_HISTORY_UNDO = `${REDUCER_KEY}/REDUCER_HISTORY_UNDO`;
export const RESIZE_ELEMENT = `${REDUCER_KEY}/RESIZE_ELEMENT`;
export const ROTATE_ELEMENTS = `${REDUCER_KEY}/ROTATE_ELEMENTS`;
export const SELECT_ELEMENT = `${REDUCER_KEY}/SELECT_ELEMENT`;
export const SELECT_ELEMENTS = `${REDUCER_KEY}/SELECT_ELEMENTS`;
export const SET_AREA_COORDINATES = `${REDUCER_KEY}/SET_AREA_COORDINATES`;
export const SET_ELEMENTS_COORDINATES = `${REDUCER_KEY}/SET_ELEMENTS_COORDINATES`;
export const SET_ELEMENTS_SCORE_TO_CURRENT_SIZE = `${REDUCER_KEY}/SET_ELEMENTS_SCORE_TO_CURRENT_SIZE`;
export const SET_ELEMENTS_SIZES = `${REDUCER_KEY}/SET_ELEMENTS_SIZES`;
export const SET_ELEMENTS_SIZES_MIN_MAX = `${REDUCER_KEY}/SET_ELEMENTS_SIZES_MIN_MAX`;
export const TOGGLE_ASPECT_RATIO = `${REDUCER_KEY}/TOGGLE_ASPECT_RATIO`;
export const UPDATE_EVENTS_STATUS = `${REDUCER_KEY}/UPDATE_EVENTS_STATUS`;
export const UPDATE_PREV_STATE = `${REDUCER_KEY}/UPDATE_PREV_STATE`;
export const UNSELECT_ELEMENT = `${REDUCER_KEY}/UNSELECT_ELEMENT`;
