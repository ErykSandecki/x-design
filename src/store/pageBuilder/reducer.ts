// others
import {
  ADD_ELEMENT,
  APPLY_ELEMENTS_INSET_TYPE,
  APPLY_ELEMENTS_SIZE_MIN_MAX_TYPE,
  CHANGE_ALIGNMENT,
  CHANGE_BACKGROUND,
  CHANGE_LAYOUT,
  CHANGE_PARENT,
  CHANGE_POSITION,
  CLEAR_PREV_STATE,
  CHANGE_LAYOUT_BOX_SIZING,
  CHANGE_LAYOUT_GRID,
  CHANGE_INSETS,
  FIT_LAYOUT,
  FLIP_ELEMENTS,
  REDUCER_HISTORY_REDO,
  REDUCER_HISTORY_SAVE,
  REDUCER_HISTORY_UNDO,
  ROTATE_ELEMENTS,
  SELECT_ELEMENT,
  SELECT_ELEMENTS,
  SET_AREA_COORDINATES,
  RESIZE_ELEMENT,
  SET_ELEMENTS_COORDINATES,
  UNSELECT_ELEMENT,
  UPDATE_EVENTS_STATUS,
  UPDATE_PREV_STATE,
  SET_ELEMENTS_SIZES,
  APPLY_ELEMENTS_SIZE_TYPE,
  SET_ELEMENTS_SIZES_MIN_MAX,
  SET_ELEMENTS_SCORE_TO_CURRENT_SIZE,
  TOGGLE_ASPECT_RATIO,
  CHANGE_LAYOUT_ALIGNMENT,
  SET_ELEMENTS_GAP,
  APPLY_ELEMENTS_GAP_TYPE,
  CHANGE_CLIP_CONTENT,
} from './actionsType';
import { BASE_PAGE } from './constants';

// types
import { AnchorResize, AnchorRotate } from './enums';
import { KeyboardKeys, TAction } from 'types';
import {
  TAddELementAction,
  TSelectElementAction,
  TPageBuilderState,
  TUnselectElementAction,
  TSelectElementsAction,
  TSetAreaCoordinatesAction,
  TUpdateEventsStatusAction,
  TResizeElementAction,
  TRotateElementsAction,
  TChangeBackgroundAction,
  TReducerHistorySaveAction,
  TChangeAlignmentAction,
  TFlipElementsAction,
  TChangeLayoutAction,
  TSetElementsCoordinatesAction,
  TSetElementsSizesAction,
  TApplyElementsSizeTypeAction,
  TSetElementsSizesMinMaxAction,
  TSetElementsScoreToCurrentSizeAction,
  TChangeLayoutAlignmentAction,
  TSetElementsGapAction,
  TApplyElementsGapTypeAction,
  TChangeLayoutBoxSizingAction,
  TChangeLayoutGridAction,
  TChangeInsetsAction,
  TApplyElementsInsetTypeAction,
  TApplyElementsSizeMinMaxTypeAction,
  TChangeClipContentAction,
} from './types';

// utils
import { filterSelectedElements } from './utils/filterSelectedElements';
import { handleAddElement } from './utils/handleAddElement';
import { handleApplyElementsGapType } from './utils/applyElementsGapType/handleApplyElementsGapType';
import { handleApplyElementsInsetType } from './utils/applyElementsInsetType/handleApplyElementsInsetType';
import { handleApplyElementsSizeMinMaxType } from './utils/applyElementsSizeMinMaxType/handleApplyElementsSizeMinMaxType';
import { handleApplyElementsSizeType } from './utils/applyElementsSizeType/handleApplyElementsSizeType';
import { handleChangeAlignment } from './utils/changeAligment/handleChangeAlignment';
import { handleChangeBackground } from './utils/handleChangeBackground';
import { handleChangeClipContent } from './utils/handleChangeClipContent';
import { handleChangeInsets } from './utils/handleChangeInsets';
import { handleChangeLayout } from './utils/changeLayout/handleChangeLayout';
import { handleChangeLayoutAlignment } from './utils/handleChangeLayoutAlignment';
import { handleChangeLayoutBoxSizing } from './utils/handleChangeLayoutBoxSizing';
import { handleChangeLayoutGrid } from './utils/changeLayoutGrid/handleChangeLayoutGrid';
import { handleChangeParent } from './utils/changeParent/handleChangeParent';
import { handleChangePosition } from './utils/handleChangePosition';
import { handleFitLayout } from './utils/handleFitLayout';
import { handleFlipElements } from './utils/handleFlipElements';
import { handleReducerHistoryRedo } from './utils/reducerHistory/handleReducerHistoryRedo';
import { handleReducerHistorySave } from './utils/reducerHistory/handleReducerHistorySave';
import { handleReducerHistoryUndo } from './utils/reducerHistory/handleReducerHistoryUndo';
import { handleResizeElement } from './utils/resizeElement/handleResizeElement';
import { handleRotateElements } from './utils/handleRotateElements';
import { handleSetElementsCoordinates } from './utils/handleSetElementsCoordinates';
import { handleSetElementsGap } from './utils/handleSetElementsGap';
import { handleSetElementsScoreToCurrentSize } from './utils/handleSetElementsScoreToCurrentSize';
import { handleSetElementsSizes } from './utils/setElementSizes/handleSetElementsSizes';
import { handleSetElementsSizesMinMax } from './utils/handleSetElementsSizesMinMax';
import { handleToggleAspectRatio } from './utils/handleToggleAspectRatio';

const initialState: TPageBuilderState = {
  currentPage: '0',
  events: {
    canMoveElements: true,
    colorSampler: false,
    draggableElements: [],
    hoverOnElement: '-1',
    isGridDropArea: false,
    isMultipleMoving: false,
    isResizing: false,
    isRotating: false,
    possibleAnchorElementId: '-1',
    possibleAnchorPosition: null,
    possibleElement: undefined,
    possibleIndexPosition: null,
    possibleParent: null,
    pressedKey: KeyboardKeys.none,
    selectedAnchorResize: AnchorResize.none,
    selectedAnchorRotate: AnchorRotate.none,
  },
  isLoading: true,
  isPending: false,
  // @ts-ignore
  pages: {
    [BASE_PAGE.id]: {
      ...BASE_PAGE,
    },
  },
};

const addElement = (
  state: TPageBuilderState,
  { payload: element }: TAction<TAddELementAction['payload']>,
): TPageBuilderState => handleAddElement(element, state);

const applyElementsGapType = (
  state: TPageBuilderState,
  { payload }: TAction<TApplyElementsGapTypeAction['payload']>,
): TPageBuilderState => handleApplyElementsGapType(payload, state);

const applyElementsInsetType = (
  state: TPageBuilderState,
  { payload }: TAction<TApplyElementsInsetTypeAction['payload']>,
): TPageBuilderState => handleApplyElementsInsetType(payload, state);

const applyElementsSizeMinMaxType = (
  state: TPageBuilderState,
  { payload }: TAction<TApplyElementsSizeMinMaxTypeAction['payload']>,
): TPageBuilderState => handleApplyElementsSizeMinMaxType(payload, state);

const applyElementsSizeType = (
  state: TPageBuilderState,
  { payload }: TAction<TApplyElementsSizeTypeAction['payload']>,
): TPageBuilderState => handleApplyElementsSizeType(payload, state);

const changeAlignment = (
  state: TPageBuilderState,
  { payload }: TAction<TChangeAlignmentAction['payload']>,
): TPageBuilderState => handleChangeAlignment(payload, state);

const changeBackground = (
  state: TPageBuilderState,
  { payload }: TAction<TChangeBackgroundAction['payload']>,
): TPageBuilderState => handleChangeBackground(payload, state);

const changeClipContent = (
  state: TPageBuilderState,
  { payload }: TAction<TChangeClipContentAction['payload']>,
): TPageBuilderState => handleChangeClipContent(payload, state);

const changeInsets = (
  state: TPageBuilderState,
  { payload }: TAction<TChangeInsetsAction['payload']>,
): TPageBuilderState => handleChangeInsets(payload, state);

const changeLayout = (
  state: TPageBuilderState,
  { payload }: TAction<TChangeLayoutAction['payload']>,
): TPageBuilderState => handleChangeLayout(payload, state);

const changeLayoutAlignment = (
  state: TPageBuilderState,
  { payload }: TAction<TChangeLayoutAlignmentAction['payload']>,
): TPageBuilderState => handleChangeLayoutAlignment(payload, state);

const changeLayoutBoxSizing = (
  state: TPageBuilderState,
  { payload }: TAction<TChangeLayoutBoxSizingAction['payload']>,
): TPageBuilderState => handleChangeLayoutBoxSizing(payload, state);

const changeLayoutGrid = (
  state: TPageBuilderState,
  { payload }: TAction<TChangeLayoutGridAction['payload']>,
): TPageBuilderState => handleChangeLayoutGrid(payload, state);

const changeParent = (state: TPageBuilderState): TPageBuilderState => handleChangeParent(state);

const changePosition = (state: TPageBuilderState): TPageBuilderState => handleChangePosition(state);

const clearPrevState = (state: TPageBuilderState): TPageBuilderState => ({
  ...state,
  pages: {
    ...state.pages,
    [state.currentPage]: {
      ...state.pages[state.currentPage],
      prevState: undefined,
    },
  },
});

const fitLayout = (state: TPageBuilderState): TPageBuilderState => handleFitLayout(state);

const flipElements = (
  state: TPageBuilderState,
  { payload }: TAction<TFlipElementsAction['payload']>,
): TPageBuilderState => handleFlipElements(payload, state);

const reducerHistoryRedo = (state: TPageBuilderState): TPageBuilderState => handleReducerHistoryRedo(state);

const reducerHistorySave = (
  state: TPageBuilderState,
  { payload }: TAction<TReducerHistorySaveAction['payload']>,
): TPageBuilderState => handleReducerHistorySave(state, payload);

const reducerHistoryUndo = (state: TPageBuilderState): TPageBuilderState => handleReducerHistoryUndo(state);

const resizeElement = (
  state: TPageBuilderState,
  { payload: { baseCoordinates, flip, height, id, mouseCoordinates, width } }: TAction<TResizeElementAction['payload']>,
): TPageBuilderState => handleResizeElement(baseCoordinates, flip, height, width, id, mouseCoordinates, state);

const rotateElements = (
  state: TPageBuilderState,
  { payload }: TAction<TRotateElementsAction['payload']>,
): TPageBuilderState => handleRotateElements(payload, state);

const selectElement = (
  state: TPageBuilderState,
  { payload: selectedElement }: TAction<TSelectElementAction['payload']>,
): TPageBuilderState => ({
  ...state,
  pages: {
    ...state.pages,
    [state.currentPage]: {
      ...state.pages[state.currentPage],
      selectedElements: filterSelectedElements(
        [...state.pages[state.currentPage].selectedElements, selectedElement],
        state,
      ),
    },
  },
});

const selectElements = (
  state: TPageBuilderState,
  { payload: selectedElements }: TAction<TSelectElementsAction['payload']>,
): TPageBuilderState => ({
  ...state,
  pages: {
    ...state.pages,
    [state.currentPage]: {
      ...state.pages[state.currentPage],
      selectedElements: filterSelectedElements(selectedElements, state),
    },
  },
});

const setAreaCoordinates = (
  state: TPageBuilderState,
  { payload: areaCoordinates }: TAction<TSetAreaCoordinatesAction['payload']>,
): TPageBuilderState => ({
  ...state,
  pages: {
    ...state.pages,
    [state.currentPage]: {
      ...state.pages[state.currentPage],
      areaCoordinates: {
        ...state.pages[state.currentPage].areaCoordinates,
        ...areaCoordinates,
      },
    },
  },
});

const setElementsCoordinates = (
  state: TPageBuilderState,
  { payload: coordinates }: TAction<TSetElementsCoordinatesAction['payload']>,
): TPageBuilderState => handleSetElementsCoordinates(coordinates, state);

const setElementsGap = (
  state: TPageBuilderState,
  { payload }: TAction<TSetElementsGapAction['payload']>,
): TPageBuilderState => handleSetElementsGap(payload, state);

const setElementsScoreToCurrentSize = (
  state: TPageBuilderState,
  { payload: { scoreType, sizeType } }: TAction<TSetElementsScoreToCurrentSizeAction['payload']>,
): TPageBuilderState => handleSetElementsScoreToCurrentSize(scoreType, sizeType, state);

const setElementsSizes = (
  state: TPageBuilderState,
  { payload: { sizeType, value } }: TAction<TSetElementsSizesAction['payload']>,
): TPageBuilderState => handleSetElementsSizes(sizeType, state, value);

const setElementsSizesMinMax = (
  state: TPageBuilderState,
  { payload: { scoreType, sizeType, value } }: TAction<TSetElementsSizesMinMaxAction['payload']>,
): TPageBuilderState => handleSetElementsSizesMinMax(scoreType, sizeType, state, value);

const toggleAspectRatio = (state: TPageBuilderState): TPageBuilderState => handleToggleAspectRatio(state);

const updateEventsStatus = (
  state: TPageBuilderState,
  { payload: events }: TAction<TUpdateEventsStatusAction['payload']>,
): TPageBuilderState => ({
  ...state,
  events: {
    ...state.events,
    ...events,
  },
});

const updatePrevState = (state: TPageBuilderState): TPageBuilderState => ({
  ...state,
  pages: {
    ...state.pages,
    [state.currentPage]: {
      ...state.pages[state.currentPage],
      prevState: state.pages[state.currentPage],
    },
  },
});

const unselectElement = (
  state: TPageBuilderState,
  { payload: id }: TAction<TUnselectElementAction['payload']>,
): TPageBuilderState => ({
  ...state,
  pages: {
    ...state.pages,
    [state.currentPage]: {
      ...state.pages[state.currentPage],
      selectedElements: state.pages[state.currentPage].selectedElements.filter((element) => element.id !== id),
    },
  },
});

const pageBuilder = (state: TPageBuilderState = initialState, action: TAction): TPageBuilderState => {
  switch (action.type) {
    case ADD_ELEMENT:
      return addElement(state, action);
    case APPLY_ELEMENTS_GAP_TYPE:
      return applyElementsGapType(state, action);
    case APPLY_ELEMENTS_INSET_TYPE:
      return applyElementsInsetType(state, action);
    case APPLY_ELEMENTS_SIZE_MIN_MAX_TYPE:
      return applyElementsSizeMinMaxType(state, action);
    case APPLY_ELEMENTS_SIZE_TYPE:
      return applyElementsSizeType(state, action);
    case CHANGE_ALIGNMENT:
      return changeAlignment(state, action);
    case CHANGE_BACKGROUND:
      return changeBackground(state, action);
    case CHANGE_CLIP_CONTENT:
      return changeClipContent(state, action);
    case CHANGE_INSETS:
      return changeInsets(state, action);
    case CHANGE_LAYOUT:
      return changeLayout(state, action);
    case CHANGE_LAYOUT_ALIGNMENT:
      return changeLayoutAlignment(state, action);
    case CHANGE_LAYOUT_BOX_SIZING:
      return changeLayoutBoxSizing(state, action);
    case CHANGE_LAYOUT_GRID:
      return changeLayoutGrid(state, action);
    case CHANGE_PARENT:
      return changeParent(state);
    case CHANGE_POSITION:
      return changePosition(state);
    case CLEAR_PREV_STATE:
      return clearPrevState(state);
    case FIT_LAYOUT:
      return fitLayout(state);
    case FLIP_ELEMENTS:
      return flipElements(state, action);
    case REDUCER_HISTORY_REDO:
      return reducerHistoryRedo(state);
    case REDUCER_HISTORY_SAVE:
      return reducerHistorySave(state, action);
    case REDUCER_HISTORY_UNDO:
      return reducerHistoryUndo(state);
    case RESIZE_ELEMENT:
      return resizeElement(state, action);
    case ROTATE_ELEMENTS:
      return rotateElements(state, action);
    case SELECT_ELEMENT:
      return selectElement(state, action);
    case SELECT_ELEMENTS:
      return selectElements(state, action);
    case SET_AREA_COORDINATES:
      return setAreaCoordinates(state, action);
    case SET_ELEMENTS_COORDINATES:
      return setElementsCoordinates(state, action);
    case SET_ELEMENTS_GAP:
      return setElementsGap(state, action);
    case SET_ELEMENTS_SCORE_TO_CURRENT_SIZE:
      return setElementsScoreToCurrentSize(state, action);
    case SET_ELEMENTS_SIZES:
      return setElementsSizes(state, action);
    case SET_ELEMENTS_SIZES_MIN_MAX:
      return setElementsSizesMinMax(state, action);
    case TOGGLE_ASPECT_RATIO:
      return toggleAspectRatio(state);
    case UPDATE_EVENTS_STATUS:
      return updateEventsStatus(state, action);
    case UPDATE_PREV_STATE:
      return updatePrevState(state);
    case UNSELECT_ELEMENT:
      return unselectElement(state, action);
    default:
      return state;
  }
};

export default pageBuilder;
