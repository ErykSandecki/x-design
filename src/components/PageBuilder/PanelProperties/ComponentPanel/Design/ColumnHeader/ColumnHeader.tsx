import { FC, memo } from 'react';
import { useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';

// components
import ColumnHeaderButtonIcons from './ColumnHeaderButtonIcons';
import { Small, UITools } from 'shared';

// others
import { translationNameSpace } from './constants';

// store
import { selectedElementsSizeSelector } from 'store/pageBuilder/selectors';

const ColumnHeader: FC = () => {
  const selectedElementsSize = useSelector(selectedElementsSizeSelector);
  const text = selectedElementsSize === 1 ? `header.frame` : `header`;
  const { t } = useTranslation();

  return (
    <UITools.SectionColumn buttonsIcon={ColumnHeaderButtonIcons(t)}>
      <Small style={{ alignItems: 'center', display: 'flex' }}>
        {t(`${translationNameSpace}.${text}`, {
          amount: selectedElementsSize,
          count: selectedElementsSize,
        })}
      </Small>
    </UITools.SectionColumn>
  );
};

export default memo(ColumnHeader);
