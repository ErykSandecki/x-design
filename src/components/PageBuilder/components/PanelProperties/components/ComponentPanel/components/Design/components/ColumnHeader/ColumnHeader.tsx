import { FC } from 'react';
import { head, size } from 'lodash';
import { useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';

// components
import { Small, UITools } from 'shared';

// others
import { translationNameSpace } from './constants';

// store
import { selectedElementsSelector } from 'store/pageBuilder/selectors';

const ColumnHeader: FC = () => {
  const selectedElements = useSelector(selectedElementsSelector);
  const amountElements = size(selectedElements);
  const text = amountElements === 1 ? `header.${head(selectedElements).type}` : `header`;
  const { t } = useTranslation();

  return (
    <UITools.SectionColumn
      buttonsIcon={[<UITools.ButtonIcon key={0} name="HtmlTag" />, <UITools.ButtonIcon key={1} name="Component" />]}
    >
      <Small>
        {t(`${translationNameSpace}.${text}`, {
          amount: amountElements,
          count: amountElements,
        })}
      </Small>
    </UITools.SectionColumn>
  );
};

export default ColumnHeader;
