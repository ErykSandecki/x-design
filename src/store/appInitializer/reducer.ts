// others
import { DEFAULT_LANGUAGE } from 'translations';
import { INIT_LANGUAGE_SUCCESS, SET_IS_APP_LOADED, SET_THEME } from './actionsType';

// types
import { TAction } from 'types';
import { TAppInitializerState, TInitLanguageSuccessAction, TSetIsAppLoadedAction, TSetThemeAction } from './types';

// utils
import { getThemePreferences } from 'utils';

const initialState: TAppInitializerState = {
  isAppLoaded: false,
  isPending: true,
  language: DEFAULT_LANGUAGE,
  theme: getThemePreferences(),
};

const initLanguageSuccess = (
  state: TAppInitializerState,
  { payload: language }: TAction<TInitLanguageSuccessAction['payload']>,
): TAppInitializerState => ({
  ...state,
  language,
});

const setIsAppLoaded = (
  state: TAppInitializerState,
  { payload: isAppLoaded }: TAction<TSetIsAppLoadedAction['payload']>,
): TAppInitializerState => ({
  ...state,
  isAppLoaded,
  isPending: false,
});

const setTheme = (
  state: TAppInitializerState,
  { payload: theme }: TAction<TSetThemeAction['payload']>,
): TAppInitializerState => ({
  ...state,
  theme,
});

const appInitializer = (state: TAppInitializerState = initialState, action: TAction): TAppInitializerState => {
  switch (action.type) {
    case INIT_LANGUAGE_SUCCESS:
      return initLanguageSuccess(state, action);
    case SET_IS_APP_LOADED:
      return setIsAppLoaded(state, action);
    case SET_THEME:
      return setTheme(state, action);
    default:
      return state;
  }
};

export default appInitializer;
