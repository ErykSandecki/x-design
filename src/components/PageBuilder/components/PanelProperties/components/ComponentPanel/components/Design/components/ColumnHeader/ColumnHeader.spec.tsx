import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import ColumnHeader from './ColumnHeader';

// mocks
import { elementMock, pageBuilderStateMock, selectedElementMock } from 'test/mocks/reducer/pageBuilderMock';

// others
import { REDUCER_KEY as PAGE_BUILDER } from 'store/pageBuilder/actionsType';

// store
import { configureStore } from 'store/store';

const currentPage = pageBuilderStateMock[PAGE_BUILDER].pages['0'];
const stateMock = {
  ...pageBuilderStateMock,
  [PAGE_BUILDER]: {
    ...pageBuilderStateMock[PAGE_BUILDER],
    pages: {
      ...pageBuilderStateMock[PAGE_BUILDER].pages,
      ['0']: {
        ...currentPage,
        elements: {
          ...currentPage.elements,
          ['-1']: {
            ...currentPage.elements['-1'],
            children: [elementMock.id],
          },
          [elementMock.id]: {
            ...elementMock,
          },
        },
        selectedElements: [selectedElementMock],
      },
    },
  },
};

describe('ColumnHeader snapshots', () => {
  it('should render ColumnHeader', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ColumnHeader />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with multiple selects', () => {
    // mock
    const store = configureStore({
      ...stateMock,
      [PAGE_BUILDER]: {
        ...pageBuilderStateMock[PAGE_BUILDER],
        pages: {
          ...pageBuilderStateMock[PAGE_BUILDER].pages,
          ['0']: {
            ...stateMock[PAGE_BUILDER].pages['0'],
            selectedElements: [selectedElementMock, { ...selectedElementMock, id: '2' }],
          },
        },
      },
    });

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ColumnHeader />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
