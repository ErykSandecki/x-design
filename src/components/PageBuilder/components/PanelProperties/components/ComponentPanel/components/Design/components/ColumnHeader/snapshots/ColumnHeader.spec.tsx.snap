// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnHeader snapshots should render ColumnHeader 1`] = `
<DocumentFragment>
  <div
    class="css-1jnxvbp SectionColumn SectionColumn--light"
    data-test-section=""
  >
    <div
      class="css-xdzdnq SectionColumn__labels SectionColumn__labels--light"
      data-test-box=""
      style="width: calc(100% - 48px - 2.5px - 8px);"
    />
    <div
      class="css-1r00qlg"
      data-test-box=""
    >
      <div
        class="css-biptg5 SectionColumn__content SectionColumn__content--light SectionColumn__content--single SectionColumn__content--single--light"
        data-test-box=""
        style="width: calc(100% - 48px - 2.5px - 8px);"
      >
        <small
          class="css-1jnxvbp Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          Frame
        </small>
        <div
          class="css-2ro7w4 SectionColumn__inputConnector SectionColumn__inputConnector--light"
          data-test-box=""
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral3 Icon--neutral3--light"
            data-test-icon="inputs-connector"
            height="24"
            name="InputsConnector"
            width="24"
          />
        </div>
      </div>
      <div
        class="css-c1qms7"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp ButtonIcon ButtonIcon--light"
          data-test-button-icon=""
          data-test-tooltip=""
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="html-tag"
            height="14"
            name="HtmlTag"
            width="14"
          />
        </div>
        <div
          class="Tooltip Tooltip--light"
        >
          <div
            class="Tooltip__content Tooltip__content--light"
          >
            Toggle ready for dev status
            <div
              class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
              data-test-icon="carrot-down"
              height="14"
              name="CarrotDown"
              width="10"
            />
          </div>
        </div>
        <div
          class="css-1jnxvbp ButtonIcon ButtonIcon--light"
          data-test-button-icon=""
          data-test-tooltip=""
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="component"
            height="14"
            name="Component"
            width="14"
          />
        </div>
        <div
          class="Tooltip Tooltip--light"
        >
          <div
            class="Tooltip__content Tooltip__content--light"
          >
            Create component
            <div
              class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
              data-test-icon="carrot-down"
              height="14"
              name="CarrotDown"
              width="10"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ColumnHeader snapshots should render with multiple selects 1`] = `
<DocumentFragment>
  <div
    class="css-1jnxvbp SectionColumn SectionColumn--light"
    data-test-section=""
  >
    <div
      class="css-xdzdnq SectionColumn__labels SectionColumn__labels--light"
      data-test-box=""
      style="width: calc(100% - 48px - 2.5px - 8px);"
    />
    <div
      class="css-1r00qlg"
      data-test-box=""
    >
      <div
        class="css-biptg5 SectionColumn__content SectionColumn__content--light SectionColumn__content--single SectionColumn__content--single--light"
        data-test-box=""
        style="width: calc(100% - 48px - 2.5px - 8px);"
      >
        <small
          class="css-1jnxvbp Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
          data-test-text=""
          style="text-align: inherit;"
        >
          2 selected
        </small>
        <div
          class="css-2ro7w4 SectionColumn__inputConnector SectionColumn__inputConnector--light"
          data-test-box=""
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral3 Icon--neutral3--light"
            data-test-icon="inputs-connector"
            height="24"
            name="InputsConnector"
            width="24"
          />
        </div>
      </div>
      <div
        class="css-c1qms7"
        data-test-box=""
      >
        <div
          class="css-1jnxvbp ButtonIcon ButtonIcon--light"
          data-test-button-icon=""
          data-test-tooltip=""
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="html-tag"
            height="14"
            name="HtmlTag"
            width="14"
          />
        </div>
        <div
          class="Tooltip Tooltip--light"
        >
          <div
            class="Tooltip__content Tooltip__content--light"
          >
            Toggle ready for dev status
            <div
              class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
              data-test-icon="carrot-down"
              height="14"
              name="CarrotDown"
              width="10"
            />
          </div>
        </div>
        <div
          class="css-1jnxvbp ButtonIcon ButtonIcon--light"
          data-test-button-icon=""
          data-test-tooltip=""
        >
          <div
            class="css-1jnxvbp Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="component"
            height="14"
            name="Component"
            width="14"
          />
        </div>
        <div
          class="Tooltip Tooltip--light"
        >
          <div
            class="Tooltip__content Tooltip__content--light"
          >
            Create component
            <div
              class="css-1jnxvbp Tooltip__carrot Tooltip__carrot--light Tooltip__carrot--topCenter Tooltip__carrot--topCenter--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
              data-test-icon="carrot-down"
              height="14"
              name="CarrotDown"
              width="10"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
